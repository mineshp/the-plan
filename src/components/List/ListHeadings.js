import React from 'react';
import PropTypes from 'prop-types';
import { Button, Dropdown, Table } from 'semantic-ui-react';

const ListItems = ({ downloadPDF, headings, listId }) => {
    const columnHeadings = [];
    headings.map((heading) => (
        columnHeadings.push(<Table.HeaderCell key={heading.id}>{heading.name}</Table.HeaderCell>)
    ));

    columnHeadings.push(
        <Table.HeaderCell key={'actions'} textAlign="right" colSpan="2">
            <Button
                as="a"
                icon="edit"
                color="green"
                href={`/list/update/${listId}`}
                title="Amend List Setup"
            />
            <Dropdown text="Export PDF" icon="file pdf outline" floating labeled button className="icon pdf">
                <Dropdown.Menu>
                    <Dropdown.Item id={listId} onClick={downloadPDF}>Save</Dropdown.Item>
                    <Dropdown.Item disabled>Email</Dropdown.Item>
                </Dropdown.Menu>
            </Dropdown>
        </Table.HeaderCell>
    );

    return (
        <Table.Header>
            <Table.Row key={123}>
                {columnHeadings}
            </Table.Row>
        </Table.Header>
    );
};

ListItems.propTypes = {
    headings: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string.isRequired,
            name: PropTypes.string.isRequired,
            position: PropTypes.number
        })
    ).isRequired,
    downloadPDF: PropTypes.func.isRequired,
    listId: PropTypes.string.isRequired
};

ListItems.defaultProps = {
    headings: []
};

export default ListItems;
