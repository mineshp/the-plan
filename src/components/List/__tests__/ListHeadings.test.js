import React from 'react';
import renderer from 'react-test-renderer';
import ListHeadings from '../ListHeadings';

const mockHeadings = [
    {
        name: 'A',
        id: '000001',
        position: 1
    },
    {
        name: 'B',
        id: '000002',
        position: 2
    }
];

describe('ListHeadings', () => {
    it('renders headings correctly', () => {
        const tree = renderer.create(<ListHeadings headings={mockHeadings} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
