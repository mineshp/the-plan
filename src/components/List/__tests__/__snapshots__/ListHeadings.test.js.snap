// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeadings renders headings correctly 1`] = `
<thead
  className=""
>
  <tr
    className=""
  >
    <th
      className=""
    >
      A
    </th>
    <th
      className=""
    >
      B
    </th>
    <th
      className="right aligned"
      colSpan="2"
    >
      <a
        className="ui green icon button"
        disabled={undefined}
        href="/list/update/112"
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        <i
          aria-hidden="true"
          className="edit icon"
        />
      </a>
      <div
        aria-busy={undefined}
        aria-disabled={undefined}
        aria-expanded={false}
        aria-multiselectable={undefined}
        className="ui button floating labeled dropdown icon pdf"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Export PDF
        </div>
        <i
          aria-hidden="true"
          className="file pdf outline icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            aria-checked={undefined}
            aria-disabled={undefined}
            aria-selected={undefined}
            className="item"
            id="112"
            onClick={[Function]}
            role="option"
          >
            Save
          </div>
          <div
            aria-checked={undefined}
            aria-disabled={true}
            aria-selected={undefined}
            className="disabled item"
            onClick={[Function]}
            role="option"
          >
            Email
          </div>
        </div>
      </div>
    </th>
  </tr>
</thead>
`;
